File: algorithms/sort/bubble_sort/bubble_sort.cpp
#include "../sort.h"

void BubbleSort(int* array, uint length, bool isDescend = false)
{
    bool isSorted;
    for (uint i = 0; i < length; i++){
        isSorted = true;
        int temp;
        for (uint j = 1; j < length - i; j++){
            if ((array[j-1] > array[j]) ^ isDescend){
                temp = array[j-1];
                array[j-1] = array[j];
                array[j] = temp;
                isSorted = false;
            }
        }
        if (isSorted)
            break;
    }
}

File: algorithms/sort/bubble_sort/bubble_sort.py
from typing import List


def bubble_sort(array: List[int], is_descend: bool = False):
    length = len(array)
    for i in range(length):
        is_sorted = True
        for j in range(1, length - i):
            if (array[j-1] > array[j]) ^ is_descend:
                array[j-1], array[j] = array[j], array[j-1]
                is_sorted = False
        if is_sorted:
            return
